esti/s3_gateway_test.go

```diff
@@ -626,3 +626,32 @@ func TestS3CopyObjectErrors(t *testing.T) {
 		require.Contains(t, err.Error(), "NoSuchKey")
 	})
 }
+
+func TestS3ReadObjectRedirect(t *testing.T) {
+	const (
+		contents = "the quick brown fox jumps over the lazy dog"
+		goodPath = "main/exists.txt"
+	)
+
+	ctx, _, repo := setupTest(t)
+	defer tearDownTest(repo)
+
+	// Upload an object
+	minioClient := newMinioClient(t, credentials.NewStaticV2)
+
+	_, err := minioClient.PutObject(ctx, repo, goodPath, strings.NewReader(contents), int64(len(contents)), minio.PutObjectOptions{})
+	if err != nil {
+		t.Errorf("PutObject(%s, %s): %s", repo, goodPath, err)
+	}
+
+	t.Run("get_exists", func(t *testing.T) {
+		opts := minio.GetObjectOptions{}
+		opts.Set("User-Agent", "s3RedirectionSupport")
+		res, err := minioClient.GetObject(ctx, repo, goodPath, opts)
+		require.NoError(t, err)
+
+		// Verify we got redirect
+		_, err = io.ReadAll(res)
+		require.Contains(t, err.Error(), "307 Temporary Redirect")
```

Comment:
Can we fetch the object here, too?

Reply:
You can read the redirect url received, no?

Reply:
You don't get the redirect url from Minio. As soon as it detects an unexpected response it transforms it into an error and returns with an empty body

Reply:
Not sure what that will achieve in the sense of the redirect mechanism. Anyway this is already tested in the spark test